lib/puma/binder.rb

```diff
@@ -5,10 +5,16 @@
 
 require 'puma/const'
 require 'puma/util'
-require 'puma/minissl/context_builder'
 require 'puma/configuration'
 
 module Puma
+
+  if HAS_SSL
```

Comment:
What's the purpose of moving these from the top of the file?

Reply:
We could move it above and use `Puma::HAS_SSL`.  I put it here for the namespace.

This was an issue I was about to post a message about.

Currently, if Puma successfully compiles with OpenSSL, it loads all the files.  Should we make so that it only loads the files if one binds to an ssl socket?

That is trickier...